components/TaskCard.vue

```diff
@@ -31,21 +63,61 @@ export default {
       default: null,
     },
   },
+  computed: {
+    ...mapGetters({ requestInProcess: 'todos/getCompleteRequest' }),
+  },
+
   data: () => ({
     task: null,
     loading: false,
   }),
+
   created() {
     this.task = _.clone(this.cardData);
   },
   methods: {
+    markDone() {
+      this.loading = true;
+      this.$store.dispatch('todos/changeTaskState', this.task).then(() => {
+        if (!this.requestInProcess) {
+          this.loading = false;
+          this.task = _.clone(this.cardData);
```

Comment:
Why do we need to clone here?

Reply:
It throws vuex  'Do not change state outside vuex' error

Reply:
Vue's reactivity does not work inside store so when the array is manipulated , the UI is not updating. I found this fix works for me. Can you suggest me a solution of this problem?

Reply:
Resolved